pkg/catalog/manager/traverse.go

```diff
@@ -273,6 +292,20 @@ func (m *Manager) traverseAndUpdate(helm *helmlib.Helm, commit string, cmt *Cata
 		errstrings = append(errstrings, fmt.Sprintf("failed to update templates. Multiple error(s) occurred: %v", updateErrors))
 	}
 	if len(errstrings) > 0 {
+		if onlyForgetErrors(forgetErrors, createErrors, updateErrors) {
+			maxTemplateNames := 5
+			forgetErrorsLen := len(forgetErrors)
+			if forgetErrorsLen < 5 {
+				maxTemplateNames = forgetErrorsLen
+			}
+			errorMessage := fmt.Sprintf("%s .... %v charts with invalid template url or name", strings.Join(errTemplateNames[:maxTemplateNames], " ; "), forgetErrorsLen)
+			setCatalogIgnoreErrorErrorState(cmt, catalog, projectCatalog, clusterCatalog, fmt.Sprintf("Error syncing catalog: %v", errorMessage))
+			if _, err := m.updateCatalogInfo(cmt, catalogType, "", false, true); err != nil {
+				return err
+			}
+			logrus.Error(fmt.Sprintf("failed to sync templates. Multiple error(s) occured: %v", forgetErrors))
+			return &controller.ForgetError{Err: errors.Errorf(fmt.Sprintf("failed to sync templates. Multiple error(s) occured: %v", forgetErrors))}
+		}
```

Comment:
So if there is a different error and an error here, this block would be skipped, is that ok ?

Reply:
yes because each `InvalidChartError` also gets added to the create and update errors, so they will get logged even if this portion gets skipped. It just ensures that if there is a weird error, we retry catalog refresh